pkg/config/config.go

```diff
@@ -232,6 +244,29 @@ func LoadFile(filename string, expandEnvVars bool, c *Config) error {
 	return LoadBytes(buf, expandEnvVars, c)
 }
 
+// Loads and merges a config from an Agent Management API.
+//  1. Read local initial config.
+//  2. Get the remote config.
+//     a) Fetch from remote. If this fails or is invalid:
+//     b) Read the remote config from cache. If this fails, return an error.
+//  4. Merge the initial and remote config into c.
+func LoadFromAgentManagementAPI(path string, expandEnvVars bool, c *Config, log *server.Logger) error {
+	// Load the initial config from disk
+	err := LoadFile(path, expandEnvVars, c)
+	if err != nil {
+		return fmt.Errorf("failed to load initial config: %w", err)
+	}
+
+	remoteConfig, err := GetRemoteConfig(filepath.Dir(path), expandEnvVars, c, log)
+	if err != nil {
+		return err
+	}
+	if err = mergo.Merge(c, remoteConfig); err != nil {
+		return fmt.Errorf("error trying to merge initial and remote configs: %w", err)
+	}
+	return nil
```

Comment:
This worries me a little bit, because reasoning about merge behaviors seems challenging. Is this what we're expecting to happen? I was under the impression that the remote config replaces the local config.

Reply:
Yeah, had a discussion and this will be replaced with a shallow, top-level merge, e.g.:
```golang
for k, v := range remoteConfig {
     c[k] = v
}
```